cmake/Macros.cmake

```diff
@@ -80,6 +88,13 @@ macro(sfml_add_library module)
     endif()
 
     set_target_warnings(${target})
+    set_public_symbols_hidden(${target})
+
+    # enable precompiled headers
+    if (SFML_ENABLE_PCH AND (NOT ${target} STREQUAL "sfml-system"))
+        message(VERBOSE "enabling PCH for SFML library '${target}'")
+        target_precompile_headers(${target} REUSE_FROM sfml-system)
+    endif()
```

Comment:
Small nitpick: `if(` doesn't have space in other places.

I like how you log the use of PCHs thoroughly, so that users can see what exactly happens!